src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRunner.cs

```diff
@@ -92,10 +93,36 @@ public static async Task<IReadOnlyList<RestoreSummary>> RunAsync(RestoreArgs res
                 restoreSummaries.Add(restoreSummary);
             }
 
+            if (restoreRequests.First().Request.ClientPolicyContext.CacheExpirationEnabled)
+            {
+                await TouchMetadata(restoreRequests, log);
+            }
+
             // Summary
             return restoreSummaries;
         }
 
+        private static async Task TouchMetadata(IEnumerable<RestoreSummaryRequest> restoreRequests, ILogger log)
+        {
+            var uniquePackageList = restoreRequests.SelectMany(restoreRequest => restoreRequest.Request.Project.TargetFrameworks.SelectMany(tfm => tfm.Dependencies)).Distinct();
+            var repoRoot = restoreRequests.First().Request.DependencyProviders.GlobalPackages.RepositoryRoot;
+
+            foreach (var package in uniquePackageList)
+            {
+                var packageMetadataFile = Path.Combine(repoRoot, package.Name, package.LibraryRange.VersionRange.OriginalString, ".nupkg.metadata");
```

Comment:
There's two major problems here:

1. `package.Name` uses the case specified in the `PackageReference`, and on filesystems that are case sensitive, this is fail if the case specified in the msbuild file does not match the path it was extracted to
2. `LibraryRange.VersionRange.originalString` won't work in a whole bunch of scenarios. Firstly, when I debugged this it was returning values like `[13.0.1, )`, meaning there wasn't a single successful file timestamp update. Secondly, it won't work for anyone using floating versions (`*`, or `6.*`, or if it does work, it'll restore the wrong version). Similarly, any restore that ends up with NU1603 will not work.

We have code somewhere that does path normalization to avoid these problems. I thought it was in `PackagePathResolver.cs`, but I don't see any `ToLowerInvariant` in there, so I guess it's somewhere else. Given that this commit clearly hasn't been tested (I'm also time limited with the number of things I need to work on. I appriciate community members spending time to help us with pull requests but pull requests that don't work or clearly introduce bugs is a waste of our time reviewing), I'm not going to hunt down the exact helper method to reuse now, but I'd start with debugging `PackageExtractor.cs` and see how the path it uses gets generated.